envi/archs/arm/disasm.py

```diff
@@ -597,10 +600,15 @@ def p_dp_reg_shift(opval, va):
 def p_mult(opval, va):
     ocode, vals = chopmul(opval)                         
     mnem, opindexes, flags = iencmul_codes.get(ocode, multfail)
+    #work around because masks match up - should be a cleaner way to do this?
+    #if Ra = 15 then smmul
     if mnem == None:
         raise envi.InvalidInstruction(
                 mesg="p_mult: invalid instruction",
                 bytez=struct.pack("<I", opval), va=va)
+    elif vals[1] == 15 and(mnem == 'smmla' or mnem == 'smmlar'):
```

Comment:
newp.  string comparison and slicing is not an option.
plus, put this before the "if mnem == None:" since that is the check for failure.  it wants to be at the end, and there's no benefit to having it down here.

create a new list like
```
iencmul_r15_codes = {
     # Basic multiplication opcodes
     binary("011101010001"): ("smmul", (0,4,2), 0),
     binary("011101010011"): ("smmulr", (0,4,2), 0),
}

```
then change this check to something like:
```
if vals[1] == 15:
    newset = iencmul_r15_codes.get(ocode)
    if newset != None:
        mnem, opindexes, flags = newset
    
```